packages/react/src/hooks/contracts/usePrepareContractWrite.ts

```diff
@@ -118,14 +119,12 @@ export function usePrepareContractWrite<
     onSuccess,
   }: UsePrepareContractWriteConfig<TAbi, TFunctionName> = {} as any,
 ) {
-  const activeChainId = useChainId()
-  const { data: signer } = useSigner<providers.JsonRpcSigner>({
-    chainId: chainId ?? activeChainId,
-  })
+  const { chain: activeChain } = useNetwork()
```

Comment:
Using chain id from `useNetwork` instead of `useChainId` since `useNetwork` returns a chain for the connected wallet (and signer), while `useChainId` returns a chain id for the `client.provider`.

Most of the time, these are the same, except when a wallet is connected to a chain not configured on the connector.